client/state/data-layer/wpcom/concierge/shifts/test/index.js

```diff
@@ -0,0 +1,76 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { exponentialBackoff } from 'state/data-layer/wpcom-http/pipeline/retry-on-failure/policies';
+import {
+	fetchConciergeShifts,
+	storeFetchedConciergeShifts,
+	showConciergeShiftsFetchError,
+} from '../';
+import {
+	CONCIERGE_SHIFTS_REQUEST,
+	CONCIERGE_SHIFTS_UPDATE,
+	NOTICE_CREATE,
+} from 'state/action-types';
+
+describe( 'wpcom-api', () => {
+	describe( 'concierge', () => {
+		test( 'fetchConciergeShifts()', () => {
+			const dispatch = jest.fn();
+			const action = {
+				type: CONCIERGE_SHIFTS_REQUEST,
+				scheduleId: 123,
+			};
+
+			fetchConciergeShifts( { dispatch }, action );
+
+			expect( dispatch ).toHaveBeenCalledWith(
+				http(
+					{
+						method: 'GET',
+						path: `/concierge/schedules/${ action.scheduleId }/shifts`,
+						apiNamespace: 'wpcom/v2',
+						retryPolicy: exponentialBackoff(),
+					},
+					action
+				)
+			);
+		} );
+
+		test( 'storeFetchedConciergeShifts()', () => {
+			const dispatch = jest.fn();
+			const mockShifts = [
+				{ description: 'shift 1' },
+				{ description: 'shift 2' },
+				{ description: 'shift 3' },
+			];
+
+			storeFetchedConciergeShifts( { dispatch }, {}, mockShifts );
+
+			expect( dispatch ).toHaveBeenCalledWith( {
+				type: CONCIERGE_SHIFTS_UPDATE,
+				shifts: mockShifts,
+			} );
+		} );
+
+		test( 'showConciergeShiftsFetchError()', () => {
+			const dispatch = jest.fn();
+
+			showConciergeShiftsFetchError( { dispatch } );
+
+			// It should be enough to make sure it dispatches a error notice action here,
+			// instead for the whole action object and the text.
+			expect( dispatch ).toHaveBeenCalledWith(
+				expect.objectContaining( {
+					type: NOTICE_CREATE,
+					notice: expect.objectContaining( {
+						status: 'is-error',
+					} ),
+				} )
+			);
```

Comment:
notice that if we want to verify the the error is created we may consider doing so in a way that doesn't hard-code the error action itself. instead, we can call `errorNotice()` here in the test itself. there's a unique id it generates but we can additionally pass that in ourselves or we can mock `impure-lodash#uniqueId` to force a test value (something >1)

Reply:
Nice catch :)
The other problem of that code was that it introduced more dependencies than the code it tested, which is a wrong design in most of the cases in my opinion. i.e. `/concierge/shifts.js` has zero knowledge of `NOTICE_CREATE` action type, but the test used it to verify its behavior. 
I've updated it in 71e445f. Since I personally don't want to repeat the same error message in both the module and the test suite, I extracted the error notice action as a exported object.

How do you think? :)

Reply:
yeah it's fine. some of the deeper question is probably how valuable the test is itself. that's not easily answerable I don't think. you did a great job either way!

Reply:
also in case you are curious, [here](https://github.com/Automattic/wp-calypso/blob/8c75edbbc9d95b2e633af4707abf7706815c75e3/client/lib/media/test/utils.js#L18) is an example of how we can mock out `impure-lodash` to force a specific message id

on the other side, we can also [force the id](https://github.com/Automattic/wp-calypso/blob/8c75edbbc9d95b2e633af4707abf7706815c75e3/client/state/data-layer/wpcom/sites/activity/test/index.js#L85)

Reply:
Ah, I didn't get it when you first mention that we can mock out `impure-lodash`. Thanks for finding the snippet for me.
I've applied it in 70d3051. Does it look good yo you? :)